lib/galaxy/model/base.py

```diff
@@ -174,18 +174,32 @@ def versioned_objects_strict(iter):
             yield obj
 
 
-if os.environ.get("GALAXY_TEST_RAISE_EXCEPTION_ON_HISTORYLESS_HDA"):
-    log.debug("Using strict flush checks")
-    versioned_objects = versioned_objects_strict  # noqa: F811
+def get_before_flush_handler():
+    if os.environ.get("GALAXY_TEST_RAISE_EXCEPTION_ON_HISTORYLESS_HDA"):
+        log.debug("Using strict flush checks")
+
+        def before_flush(session, flush_context, instances):
+            for obj in session.new:
```

Comment:
Isn't that what `versioned_objects_strict()` does? If so, why not do something like this:
```py
for obj in versioned_objects_strict(session.new):
    pass
```

Reply:
It creates a version, which doesn't make sense for new objects,

Reply:
And then, if I'm not mistaken, you wouldn't need this `if/then` that  checks `os.environ` and contains 2 similar function definitions, and just use the existing logic, I think? i.e., setting `versioned_objects = versioned_objects_strict` would still work?

Reply:
I don't think one is better than the other with the exception this one is slightly better since we don't run the os.environ check in the module space ?